### PR TITLE
Reuse runJob for Delete and Read

### PR DESCRIPTION
**Pull request by ygalblum**
## Type of change

- [x] Refactor

## Description

Rename executionMode to unlink from patch
Move runJob and its internals to utiles  and use in Read and Delete jobs
Add ObjectFinalizer in addition to ItemHandler
Set Job for Executor and make sure it is not accesses once it is passed to executor
Remove redundant labelSelector and patchType fields from object

### CODE REVIEW

**Comment by ygalblum** [comment]
@rsevilla87 @vishnuchalla This is a first stage in some code consolidation work that I did in preparation of adding new functionality

File: pkg/burner/job.go [hunk 1]
```diff
@@ -343,19 +349,18 @@ func newExecutorList(configSpec config.Spec, kubeClientProvider *config.KubeClie
 			log.Fatalf("Unknown jobType: %s", job.JobType)
 		}
 		for _, j := range executorList {
-			if job.Name == j.Job.Name {
-				log.Fatalf("Job names must be unique: %s", job.Name)
+			if ex.Name == j.Name {
+				log.Fatalf("Job names must be unique: %s", ex.Name)
 			}
```
Comment (rsevilla87) [inline pkg/burner/job.go:354]: We can remove this conditional, just realized that this is checked by the jobIsDuped() function in config/config.go

File: pkg/burner/job.go [hunk 2]
```diff
@@ -43,23 +44,26 @@ import (
...
+	gvr        schema.GroupVersionResource
+	objectSpec []byte
+	kind       string
+	namespace  string
 	config.Object
 }
 
 // Executor contains the information required to execute a job
+type ItemHandler func(ex *Executor, obj object, originalItem unstructured.Unstructured, iteration int, wg *sync.WaitGroup)
+type ObjectFinalizer func(obj object)
```
Comment (rsevilla87) [inline pkg/burner/job.go:56]: To we actually need these two new types?

File: pkg/burner/delete.go [hunk 3]
```diff
@@ -47,72 +58,47 @@ func setupDeleteJob(jobConfig config.Job) Executor {
 			log.Fatalf("Empty labelSelectors not allowed with: %s", o.Kind)
 		}
 		obj := object{
-			gvr:           mapping.Resource,
-			labelSelector: o.LabelSelector,
```
Comment (rsevilla87) [inline pkg/burner/delete.go:51]: Interesting... seems we can use the LabelSelector attribute directly from the config/object, rather than re-defining it

Comment (ygalblum) [reply pkg/burner/delete.go:51]: Yes, since executor extends the Job it can access all fields directly.

File: pkg/burner/job.go (hunk 2 above)
Comment (ygalblum) [reply pkg/burner/job.go:56]: These types are used in the `Executor` struct below. I think it's cleaner and easier to read than just adding the method signature in the struct's definition.

File: pkg/burner/job.go [hunk 4]
```diff
@@ -141,8 +152,7 @@ func Run(configSpec config.Spec, kubeClientProvider *config.KubeClientProvider,
 			measurements.SetJobConfig(&job.Job, kubeClientProvider)
 			log.Infof("Triggering job: %s", job.Name)
 			measurements.Start()
-			switch job.JobType {
-			case config.CreationJob:
+			if job.JobType == config.CreationJob {
```
[the remark this answers is not in the document]
Comment (ygalblum) [reply pkg/burner/job.go:155]: Yes, the create operation is different than the rest and that's why I did not address it yet. We can address it later

File: pkg/burner/job.go (hunk 1 above)
Comment (ygalblum) [reply pkg/burner/job.go:354]: Done

**Review by rsevilla87 — APPROVED** [review]
lgtm

**Pull request merged**